xbmc/pvr/PVRManager.cpp

```diff
@@ -351,16 +351,30 @@ void CPVRManager::ResetProperties()
   m_knownClients.clear();
 }
 
+namespace
+{
+class CClientsInitThread : private CThread
+{
+public:
+  explicit CClientsInitThread(const std::shared_ptr<CPVRClients>& clients)
+    : CThread("PVRClientsInitThread"), m_clients(clients)
+  {
+    Create(true);
+  }
+
+private:
+  void Process() override { m_clients->Start(); }
```

Comment:
Doesn't `Process()` call `delete this;`? Even if not, the question raises flags.

Reply:
Yes, and 'this' is not the pvr manager instance, but the client start thread and this is okay. It's a one-shot action. When done, the thread dies.

Reply:
My point is, does the override need to call the base implementation for that behavior?

Reply:
> does the override need to call the base implementation for that behavior?

No, because in the ctor of the derived class `Create(bool bAutoDelete)` is called with param `bAutoDelete`set to `true`.
See also: https://github.com/xbmc/xbmc/blob/master/xbmc/threads/Thread.h#L52